fvtest/compilertriltest/CompareTest.cpp

```diff
@@ -379,3 +379,92 @@ INSTANTIATE_TEST_CASE_P(CompareTest, UInt64Compare, ::testing::Combine(
         std::make_tuple("lucmpge", lucmpge),
         std::make_tuple("lucmplt", lucmplt),
         std::make_tuple("lucmple", lucmple) )));
+
+int32_t ificmpeq(int32_t l, int32_t r) {
+    return (l == r) ? 123 : -456;
+}
+
+int32_t ificmpne(int32_t l, int32_t r) {
+    return (l != r) ? 123 : -456;
+}
+
+int32_t ificmplt(int32_t l, int32_t r) {
+    return (l < r) ? 123 : -456;
+}
+
+int32_t ificmple(int32_t l, int32_t r) {
+    return (l <= r) ? 123 : -456;
+}
+
+int32_t ificmpge(int32_t l, int32_t r) {
+    return (l >= r) ? 123 : -456;
+}
+
+int32_t ificmpgt(int32_t l, int32_t r) {
+    return (l > r) ? 123 : -456;
+}
+
+class Int32IfCompare : public TRTest::BinaryOpTest<int32_t> {};
+
+TEST_P(Int32IfCompare, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[256] = {0};
+    std::snprintf(inputTrees, 256,
+       "(method return=Int32 "
+         "(block "
+           "(%s target=b1 (iconst %d) (iconst %d))) "
+         "(block (ireturn (iconst -456))) "
+         "(block name=b1 (ireturn (iconst 123)))"
+       ")",
+       param.opcode.c_str(), param.lhs, param.rhs);
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int32_t (*)(int32_t, int32_t)>();
+    volatile auto exp = param.oracle(param.lhs, param.rhs);
+    volatile auto act = entry_point(param.lhs, param.rhs);
```

Comment:
```suggestion
    volatile auto act = entry_point();
```

This is a `UsingConst` variant. :wink: